PermissionVault/contracts/PermissionVault.sol

```diff
@@ -0,0 +1,366 @@
+
+  /* -------------------------------------------------------------------------- */
+  /*                                   Events                                   */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * Emitted when new controller was added
+   */
+  event AddController(address controller);
+  /**
+   * Emitted when controller was removed
+   */
+  event RemoveController(address controller);
+
+  /**
+   * Emitted when controller deposited Ether
+   */
+  event DepositEther(address controller, uint256 amount);
+  /**
+   * Emitted when controller withdrawed Ether
+   */
+  event WithdrawEther(address controller, address to, uint256 amount);
+
+  /**
+   * Emitted when controller deposited ERC20
+   */
+  event DepositERC20(address controller, IERC20 token, uint256 amount);
+  /**
+   * Emitted when controller withdrawed ERC20
+   */
+  event WithdrawERC20(
+    address controller,
+    address to,
+    IERC20 token,
+    uint256 amount
+  );
+
+  /**
+   * Emitted when controller deposited ERC721
+   */
+  event DepositERC721(address controller, IERC721 token, uint256 id);
+  /**
+   * Emitted when controller withdrawed ERC721
+   */
+  event WithdrawERC721(
+    address controller,
+    address to,
+    IERC721 token,
+    uint256 id
+  );
+
+  /**
+   * Emitted when controller deposited ERC1155
+   */
+  event DepositERC1155(
+    address controller,
+    IERC1155 token,
+    uint256 id,
+    uint256 amount
+  );
+  /**
+   * Emitted when controller withdrawed ERC1155
+   */
+  event WithdrawERC1155(
+    address controller,
+    address to,
+    IERC1155 token,
+    uint256 id,
+    uint256 amount
+  );
+
+  /* -------------------------------------------------------------------------- */
+  /*                                  Modifiers                                 */
+  /* -------------------------------------------------------------------------- */
+
+  modifier onlyController() {
+    if (!hasRole(CONTROLLER_ROLE, msg.sender)) {
+      revert NotController();
+    }
+    _;
+  }
+
+  /* -------------------------------------------------------------------------- */
+  /*                             External Functions                             */
+  /* -------------------------------------------------------------------------- */
+
+  /**
+   * @dev Constructor
+   */
+  constructor() {
+    _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    _setupRole(CONTROLLER_ROLE, msg.sender);
+  }
+
+  /**
+   * @notice The owner can use this function to pause or unpause the contract.
+   * If newPaused is true, the contract will be paused, and no functions can be
+   * called except setPaused. If newPaused is false, the contract will be
+   * unpaused.
+   * @dev Callable by owner
+   * @param newPaused Flag to new paused state
+   */
+  function setPaused(bool newPaused) external onlyOwner {
+    if (newPaused) {
+      _pause();
+    } else {
+      _unpause();
+    }
+  }
```

Comment:
this contract inherited Pausable contract and has the setPaused function.
but you didn't use that in anywhere. 
please add the modifier for functions. or remove Pausable.

Reply:
Good point, thank you for letting me know